usecase/interactor/task_interactor.go

```diff
@@ -37,3 +43,75 @@ func (ti *taskInteractor) GetOne(t *model.Task, id uint) (*model.Task, error) {
 
 	return ti.TaskPresenter.ResponseTask(t), nil
 }
+
+func (ti *taskInteractor) GetConcurrently(t []*model.Task, onlyEven bool, items, itemsPerWorker int) ([]*model.Task, error) {
+	allTasks, err := ti.TaskRepository.FindAll(t) // Load all the Tasks in Memory
+	if err != nil {
+		return nil, err
+	}
+
+	t = []*model.Task{}                            // Slice which will contain filtered tasks
+	tasks := make(chan *model.Task, len(allTasks)) // Channel with size of all tasks
+	results := make(chan *model.Task, items)       // Channel of results with size of maximum number of items
+
+	// Calculate number of workers and run them
+	numberOfWorkers := int(math.Ceil(float64(items) / float64(itemsPerWorker)))
+
+	wg.Add(numberOfWorkers) // Add all the workers to WaitGroup
+
+	for w := 1; w <= numberOfWorkers; w++ {
+		go worker(w, itemsPerWorker, onlyEven, tasks, results)
+	}
+	fmt.Println(numberOfWorkers, "workers started")
+
+	for _, task := range allTasks { // Send all tasks to Channel
+		tasks <- task
+	}
+	close(tasks)
+	fmt.Printf("All tasks were send to channel\n")
+
+	go func() { // Wait all the workers finish to close results channels
```

Comment:
Why is this in a separate goroutine?

Reply:
Actually I found that way of wait in Stack Overflow 😅
The main benefit I saw is that the program starts to read the results before all the workers finishes, probably in this case it's not a big time save, but if we would have many items to process and each item would need more time to be processed we would see the benefit of waiting in a goroutine while parsing results